### PR TITLE
Update and standardise exceptions handling

### LINKED ISSUE #176: Update Person toString method
Currently is showing "Other contact" in Messages

### PR DESCRIPTION
**Pull request by chiageng**
Fixes #176 

### New Functionality
- **Verify Missing Prefixes:** Introduced a new function `verifyMissingPrefixes` for missing prefix verification.
- **Enhanced Unknown Prefix Verification:** Revised `verifyUnknownPrefixes` to efficiently manage both unknown prefix identification and exception handling.

### Exception Handling
- **Standardization Across CommandParsers:** Implemented a uniform approach to exception handling within every `CommandParser` utilizing the newly introduced `verifyMissingPrefixes` and updated `verifyUnknownPrefixes` functions.

### Clear command
- **Change Format of Clear Command.** /clear to use clear command
- **Allow Undo Command Works on Clear Command.** Save a new state in address book when executed /clear

### Code Refactor
- **Unified Message Formatting:** Refactored message-related functionalities by abstracting out the formatting logic into a new function, `formatPerson`. This affects:
  - AddMessages
  - EditMessages
  - PinMessages
  - UnpinMessages
  ... and others.

### Testing
- Updated all test cases to align with the new exception handling process and the centralized message formatting logic.

### Logging
- Integrated automatic logging of warnings within both `verifyMissingPrefixes` and `verifyUnknownPrefixes` functions for enhanced debugging and monitoring.

### CODE REVIEW

File: src/main/java/seedu/address/logic/messages/AddMessages.java [hunk 1]
```diff
@@ -15,21 +10,5 @@ public class AddMessages extends Messages {
...
-            builder.append("Supplier ");
-        } else if (person instanceof Maintainer) {
-            builder.append("Maintainer ");
-        } else {
-            builder.append("Other Contact ");
-        }
-        builder.append(person.getName());
-        return builder.toString();
-    }
+
```
Comment (yleeyilin) [inline src/main/java/seedu/address/logic/messages/AddMessages.java:13]: Extra line here

File: src/main/java/seedu/address/logic/messages/EditMessages.java [hunk 2]
```diff
@@ -16,6 +11,9 @@ public class EditMessages extends Messages {
             + "Edit requires a field prefix. %1$s \uD83D\uDC3E";
     public static final String MESSAGE_EDIT_MISSING_NAME = "Failed to edit Pooch Contact - "
             + "Edit requires a name field. %1$s \uD83D\uDC3E";
+
```
Comment (yleeyilin) [inline src/main/java/seedu/address/logic/messages/EditMessages.java:14]: Extra line here

**Review by yleeyilin — APPROVED** [review]
lgtm, except some minor comments

File: src/main/java/seedu/address/logic/messages/EditMessages.java [hunk 3]
```diff
@@ -16,6 +11,9 @@ public class EditMessages extends Messages {
...
+    public static final String MESSAGE_EDIT_NAME = "Failed to edit Pooch Contact - "
+            + "Edit detect duplicate name fields. %1$s \uD83D\uDC3E";
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/messages/EditMessages.java:16]: detected

**Review by jamessinmaojun — APPROVED** [review]
lgtm

**Pull request merged**